src/js/yeldaWebchatInjector.js

```diff
@@ -365,22 +365,26 @@ class YeldaChat {
   }
 
   /**
-   * Handles communcation between parent window and iframe, mainly for open and closing the chat
+   * Handles communication between parent window and iframe, mainly for open and closing the chat
    * handles also the chat bubble style
    * @param {event} event
    */
   messageListener(event) {
-    if (event.data === 'closeChat' || event.message === 'closeChat') {
+    if (!event.data) {
+      return
+    }
+
+    if (event.data === 'closeChat') {
       this.closeChat()
       return
     }
 
-    if (event.data === 'openChat' || event.message === 'openChat') {
+    if (event.data === 'openChat') {
       this.openChat()
       return
     }
 
-    if (event.data && event.data.event && event.data.event === 'openLightGallery') {
```

Comment:
if would move here abstainLeaveViewport and handle simple events (string data) and complex ones (object) separately
```js
     /*
     * Simple event management parent.postMessage('string message')
     */
     if(!event.data.event) {
       switch (event.data) {
         case confing.EVENT_TYPES.OPEN_CHAT:
           this.openChat()
         break
          case confing.EVENT_TYPES.CLOSE_CHAT:
           this.closeChat()
         break
         case confing.EVENT_TYPES.ABSTAIN_LEAVE_VIEWPORT:
           // Triggered from webchat to stop listening to the leave viewport event
           this.listenLeaveViewport(true)
         break
        default:
         return 
      }
    return
  }  
   
    /*
     * Complex event management parent.postMessage({ event: 'xxxx', data: yyyy })
     */
  switch (event.data.event) {
    case confing.EVENT_TYPES.OPEN_LIGHT_GALLERY:
      if (!event.data.mediaSources || !event.data.mediaSources.length) {
        return
      }
      this.handleLightGallery(event.data)
      break
    case confing.EVENT_TYPES.IS_SENDING_MESSAGE:
 /**
     * Custom event to send to Yelda to keep track of the webchat ability to receive a new message
     * If isSendingMessage data is true, it means that a user message has already been sent to the webchat and
     * is still waiting for an answer
     * If isSendingMessage data is false, the webchat is ready to receive new user messages
     */
      if (event.data.hasOwnProperty('data')) {
        window.dispatchEvent(new CustomEvent('isSendingMessage', { detail: event.data.data }))
      }
      break
    case confing.EVENT_TYPES.LISTEN_LEAVE_VIEWPORT:
      // Triggered from webchat to listen leave viewport event
      this.pushNotificationId = event.data.pushNotificationId
      this.listenLeaveViewport()
      break
    case confing.EVENT_TYPES.LISTEN_LEAVE_VIEWPORT:
    // Triggered from webchat to stop listening to the leave viewport event
      this.listenLeaveViewport(true)
      break
    default:
      return 
   }
 }